src/scheduler/simulate.c

```diff
@@ -1547,54 +1538,34 @@ add_timed_event(timed_event *events, timed_event *te)
  *
  * @param[in] sinfo    - sinfo which contains calendar to delete from
  * @param[in] e        - event to delete
- * @param[in] flags    - flag bitfield
- *						 DE_UNLINK - unlink event and don't free
  *
- * @return int
- * @retval 1 event was successfully deleted
- * @retval 0 failure
+ * @return void
  */
 
-int
-delete_event(server_info *sinfo, timed_event *e, unsigned int flags)
+void
+delete_event(server_info *sinfo, timed_event *e)
 {
-	timed_event *cur_e;
-	timed_event *prev_e = NULL;
 	event_list *calendar;
 
 	if (sinfo == NULL || e == NULL)
-		return 0;
+		return;
 
 	calendar = sinfo->calendar;
 
+	if (calendar->next_event == e)
+		calendar->next_event = e->next;
 
-	for (cur_e = calendar->events; cur_e != e && cur_e != NULL;
-		prev_e = cur_e, cur_e = cur_e->next)
-			;
-
-	/* found our event to delete */
-	if (cur_e != NULL) {
-		if (calendar->next_event == cur_e)
-			calendar->next_event = cur_e->next;
-
-		if (prev_e == NULL)
-			calendar->events = cur_e->next;
-		else
-			prev_e->next = cur_e->next;
-
-		if ((flags & DE_UNLINK) == 0) {
-			if (cur_e->event_type & TIMED_RUN_EVENT)
-				((resource_resv *)cur_e->event_ptr)->run_event = NULL;
-			else if (cur_e->event_type & TIMED_END_EVENT)
-				((resource_resv *)cur_e->event_ptr)->run_event = NULL;
-
-			free_timed_event(cur_e);
-		}
-
-		return 1;
+	if (e->prev == NULL) {
+		calendar->events = e->next;
+		if (e->next != NULL)
+			e->next->prev = NULL;
+	} else {
+		e->prev->next = e->next;
+		if (e->next != NULL)
```

Comment:
you can move this condition and assignment out of the if/else block. because effectively you are doing the same operation in both places, its just that e->prev is NULL in the 'if' block.